modules/match.js

```diff
@@ -1,10 +1,7 @@
 import invariant from 'invariant'
-import createMemoryHistory from 'history/lib/createMemoryHistory'
-import useBasename from 'history/lib/useBasename'
+import createMemoryHistory from './createMemoryHistory'
+import createTransitionManager from './createTransitionManager'
 import { createRoutes } from './RouteUtils'
-import useRoutes from './useRoutes'
-
-const createHistory = useRoutes(useBasename(createMemoryHistory))
```

Comment:
There's a good use case here: rackt/history#188 Swapping out a translation based on a base URL.

 I've actually been meaning to do the same thing myself, as a I have widget app that will be included in the context of many different URLs. Right now, I have a `createRoutes(baseURL)`  function that runs every time, which is a pretty ugly hack.